frontend/src/utils/getAllCourseOfferings.ts

```diff
@@ -0,0 +1,59 @@
+import axios from 'axios';
+import { CourseLegacyOfferings, PlannerCourse, Term } from 'types/planner';
+
+export type ManyCoursesOfferings = {
+  [course: string]: CourseLegacyOfferings;
+};
+
+type APIResult = {
+  terms: {
+    [year: string]: Term[] | null;
+  };
+  fails: unknown[];
+};
+
+const getAllCourseOfferings = async (
+  courses: string[],
+  years: string[]
+): Promise<ManyCoursesOfferings> => {
+  if (courses.length === 0 || years.length === 0) {
+    return {};
+  }
+
+  const yearsStr = years.join('+');
+  const offers: ManyCoursesOfferings = {};
+
+  await Promise.all(
+    courses.map(async (course) => {
+      try {
+        const res = await axios.get<APIResult>(`/courses/termsOffered/${course}/${yearsStr}`);
+        if (res.status === 200) {
+          let mostRecentYear: Term[] = [];
+          offers[course] = {};
+          Object.entries(res.data.terms).forEach(([year, terms]) => {
+            offers[course][year] = terms ?? mostRecentYear;
+            if (terms !== null) {
+              mostRecentYear = terms;
+            }
+          });
```

Comment:
The route has been updated to use the `LIVE_YEAR` as a fallback  - which already isn't perfect but, it's something to keep in mind. the difference between this and just using the be's implementation is that this will also predict the future for courses that aren't being offered in this term. I feel that perhaps, we should add a future card to detect if a course has been offered inconsistently and just throw up a tooltip / warning for those courses.

Thoughts?

Reply:
i like